packages/analytics/client/src/analytics_client/context_service.ts

```diff
@@ -108,3 +107,7 @@ export class ContextService {
     }, {} as Partial<EventContext>);
   }
 }
+
+function isObject(value: unknown): value is Record<string, unknown> {
+  return typeof value === 'object' && value !== null;
+}
```

Comment:
Not sure it's worth the trouble, but what do you think about moving this to shared lib like `@kbn/std`?

Reply:
I thought about it. But we're thinking about porting this package to an independent NPM module. So I thought it might be better not to import other packages at this stage.